python-package/lightgbm/basic.py

```diff
@@ -122,7 +120,17 @@ def _load_lib() -> Optional[ctypes.CDLL]:
     return lib
 
 
-_LIB = _load_lib()
+# we don't need lib_lightgbm while building docs
+_LIB: ctypes.CDLL
+if environ.get('LIGHTGBM_BUILD_DOC', False):
+    class _MockCDLL:
+        def __getattr__(self, name):
+            def _catchall(*args, **kwargs):
+                return None
+            return _catchall
```

Comment:
Can we use bult-in `[Mock`](https://docs.python.org/3/library/unittest.mock.html#unittest.mock.Mock) instead?
```python
_MockCDLL = Mock(ctypes.CDLL)
```

Reply:
😱 I didn't know you could pass an existing class to `Mock()`!! I've only ever instantiated empty `Mock()` objects in tests.

Thanks for teaching me that, totally agree!

Updated in https://github.com/microsoft/LightGBM/pull/5357/commits/abab6baf5eec2a62a428a8e53a11982771f63806.

I'm proposing putting the `unittest.mock` import inside the `if` statement checking if we are doing docs-building so that the small memory usage and latency incurred by that can be avoided for most uses of `lightgbm`. I'll move that import back to the top if you disagree with that decision...just wanted to say that I did this intentionally, and why.